### PR TITLE
Content creators should be able to delete questions in the assessment…

### PR DESCRIPTION
**Pull request by mehalshah**
… creation screen

### CODE REVIEW

**Comment by mehalshah** [comment]
image

File: code-studio/src/css/plc/content_creation [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
     input, select {
       margin: auto;
+      width: 90%;
```
Comment (joshlory) [inline code-studio/src/css/plc/content_creation:21]: Does this file not have an extension?

File: dashboard/app/views/plc/course_units/show.html.haml [hunk 2]
```diff
@@ -36,6 +36,8 @@
             = question.question
           %td
           %td
+          %td
+            = link_to '', question, :method => :delete, :data => { :confirm => 'Are you sure you want to delete this question?' }, class: 'fa fa-trash btn'
```
Comment (joshlory) [inline dashboard/app/views/plc/course_units/show.html.haml]: Nit: use [Ruby 1.9 hash syntax](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#ruby-hashes).

File: dashboard/test/controllers/plc/evaluation_questions_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require 'test_helper'
+
+class Plc::EvaluationQuestionsControllerTest < ActionController::TestCase
+  setup do
+    user = create (:admin)
```
Comment (joshlory) [inline dashboard/test/controllers/plc/evaluation_questions_controller_test.rb]: Nit: extra space after `create` changes the meaning.  It has the same result with one argument, but adding a second argument would throw an exception.

File: dashboard/test/controllers/plc/evaluation_questions_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    sign_in(user)
+    course = create(:plc_course)
+    @course_unit = create(:plc_course_unit, plc_course: course)
+    @plc_evaluation_question = create(:plc_evaluation_question, plc_course_unit: @course_unit)
+  end
+
+  test "should destroy plc_evaluation_question" do
+    assert_difference('Plc::EvaluationQuestion.count', -1) do
```
Comment (joshlory) [inline dashboard/test/controllers/plc/evaluation_questions_controller_test.rb:13]: A better test might be to make sure this specific `EvaluationQuestion` was destroyed, not just that the count decreased.

**Comment by joshlory** [comment]
One general note for the PLC tools: I always forget where the entry point is because of the nonstandard naming.  E.g. `/plc` and `/plc/tasks` both are 404.  Getting rid of all the index views makes the tools difficult to navigate.  I have to remember to go to `/plc/content_creator/show_courses_and_modules`.

Can the root content be located at `/plc`, or can `/plc` redirect here?

File: code-studio/src/css/plc/content_creation (hunk 1 above)
Comment (mehalshah) [inline code-studio/src/css/plc/content_creation:21]: Hmm - looks like when I created the scss files in the plc dir I didn't put extensions. I can do that

File: dashboard/app/views/plc/course_units/show.html.haml (hunk 2 above)
Comment (mehalshah) [inline dashboard/app/views/plc/course_units/show.html.haml]: Will do

File: dashboard/test/controllers/plc/evaluation_questions_controller_test.rb (hunk 3 above)
Comment (mehalshah) [inline dashboard/test/controllers/plc/evaluation_questions_controller_test.rb]: Will fix

File: dashboard/test/controllers/plc/evaluation_questions_controller_test.rb (hunk 4 above)
Comment (mehalshah) [inline dashboard/test/controllers/plc/evaluation_questions_controller_test.rb:13]: Will change

**Comment by mehalshah** [comment]
Let me think this over - we can absolutely do the redirection, I'm just trying to figure out where teachers should get redirected to

**Comment by joshlory** [comment]
How do I get to the Evaluation Questions view in the screenshot above?

**Comment by joshlory** [comment]
Not related to this change, but you're trying to load the `IconLibrary` on every page, which throws this error in the console:
```
Error: _registerComponent(...): Target container is not a DOM element.(…)
```

**Comment by joshlory** [comment]
Done with my first pass.

File: dashboard/app/views/plc/course_units/show.html.haml [hunk 5]
```diff
@@ -37,7 +37,7 @@
           %td
           %td
           %td
-            = link_to '', question, :method => :delete, :data => { :confirm => 'Are you sure you want to delete this question?' }, class: 'fa fa-trash btn'
+            = link_to '', question, method: :delete, data: { :confirm => 'Are you sure you want to delete this question?' }, class: 'fa fa-trash btn'
```
Comment (joshlory) [inline dashboard/app/views/plc/course_units/show.html.haml:40]: `:confirm =>` should be `confirm:`

File: dashboard/config/routes.rb [hunk 6]
```diff
@@ -285,6 +285,11 @@ def redirect_to_teacher_dashboard
     concerns :ops_routes
   end
 
+  get '/plc/content_creator/show_courses_and_modules', to: 'plc/content_creator#show_courses_and_modules'
+  ['courses', 'learning_modules', 'tasks', 'course_units', 'evaluation_questions'].each do |object|
+    get '/plc/' + object, to: redirect('plc/content_creator/show_courses_and_modules')
+  end
```
Comment (joshlory) [inline dashboard/config/routes.rb:291]: I think this is okay for now.  But we should think through what we want the final link structure to be before release, when people will start bookmarking URLs and they become hard to change.

File: dashboard/app/views/plc/course_units/show.html.haml (hunk 5 above)
Comment (mehalshah) [inline dashboard/app/views/plc/course_units/show.html.haml:40]: Done

**Pull request merged**